### PR TITLE
Replace Cardano.API.Typed import with Cardano.Api

### PR DESCRIPTION
**Pull request by Jimbo4350**
- Rename `Cardano.API` to `Cardano.Api`
- Replace `Cardano.API.Typed` with `Cardano.Api`
- Additionally import `Cardano.Api.Byron` or `Cardano.Api.Shelley` as necessary

### CODE REVIEW

**Review by dcoutts — APPROVED** [review]
LGTM!

**Comment by dcoutts** [comment]
bors merge

File: cardano-api/ChangeLog.md [hunk 1]
```diff
@@ -35,7 +35,7 @@ None
 ## 1.21.2 -- September 2020
 
 - Add a Ed25519-BIP32 instance of the new crypto classes (#1933, #1952)
-- Adjust what is exposed via Cardano.API.{Byron,Shelley} (#1932)
+- Adjust what is exposed via Cardano.Api.{Byron,Shelley} (#1932)
```
Comment (intricate) [inline cardano-api/ChangeLog.md:38]: We shouldn't change the changelogs, right?

Comment (dcoutts) [reply cardano-api/ChangeLog.md:38]: I don't think it matters too much in this case, it might even be helpful.

**Pull request merged**